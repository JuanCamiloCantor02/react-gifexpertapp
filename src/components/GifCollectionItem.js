import React from 'react'
import PropTypes from 'prop-types';

export const GifCollectionItem = ({id, title, url}) => {
   
   
    return (
        <div className="card animate__animated animate__backInUp">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifCollectionItem.propTypes ={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}