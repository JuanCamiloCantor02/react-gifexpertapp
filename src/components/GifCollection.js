import React/* , { useEffect, useState } */ from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
/* import { getCollection } from '../helpers/getGifs'; */
import { GifCollectionItem } from './GifCollectionItem';

export const GifCollection = ({category}) => {
   
    /* const [images, setImages] = useState([]);
    
    useEffect(() => {
        getCollection(category)
            .then(setImages);
    }, [category]) */

    const {data:images, loading} = useFetchGifs(category);

    


    return (
        <>
            <h3 className="animate__animated animate__pulse">{category}</h3>

            {loading && <p>loading...</p>}
            
            <div className="card-collection">
                
                    {
                        images.map( img =>(
                        <GifCollectionItem
                            key={img.id} 
                            {...img} 
                        />
                        ))
                    }
                
            </div>
        </>
    )
}
