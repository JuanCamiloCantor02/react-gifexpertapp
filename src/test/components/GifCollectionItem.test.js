import React from 'react'
import {shallow} from "enzyme";
import { GifCollectionItem } from '../../components/GifCollectionItem';

describe('Preubas en <GifCollectionItem/>', () => {

    const title = "Hola amigos"
    const url = "https://localhost/amigo.jpg"
    
    const wrapper = shallow(<GifCollectionItem title={title} url={url}/>)

    test('Debe mostrar el componente correctamente ', () => {
        

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);

    })

    test('debe tener la imagen  igual al url y alt de los props ', () => {
        
        const img = wrapper.find("img");
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
        

    })

    test('debe tener animate__backInUp', () => {
        const div = wrapper.find("div");
        const className = div.prop("className");
        

        expect(className.includes("animate__backInUp")).toBe(true);

    })
    
    
    
    
})
