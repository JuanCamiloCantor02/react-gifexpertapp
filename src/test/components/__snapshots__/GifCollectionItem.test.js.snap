// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preubas en <GifCollectionItem/> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__backInUp"
>
  <img
    alt="Hola amigos"
    src="https://localhost/amigo.jpg"
  />
  <p>
    Hola amigos
  </p>
</div>
`;
